### PR TITLE
Make parseLog more approachable.

### PR DESCRIPTION
**Pull request by shaun-whitely**
- Provide the start of a suggested implementation

### CODE REVIEW

File: src/main/scala/fundamentals/level03/TryExercises.scala [hunk 1]
```diff
@@ -86,6 +86,20 @@ object TryExercises {
...
 
+  /**
+    * Write a function that converts a `Try[A]` to `Option[A]`.
+    *
+    * scala> tryToOption(parseIntSafe("1"))
+    * = Some(1)
+    * scala> tryToOption(parseIntSafe("abc"))
+    * = None
+    */
+  def tryToOption[A](tryA: Try[A]): Option[A] =
```
Comment (wjlow) [inline src/main/scala/fundamentals/level03/TryExercises.scala]: They would've learned `.toOption` by this point. Perhaps remove the pattern match so they can solve it however they want.

File: src/main/scala/fundamentals/level04/LogParser.scala [hunk 2]
```diff
@@ -50,6 +50,15 @@ object LogParser {
    * 2. Comment out the contents of src/test/scala/fundamentals/level04/Types.scala
    */
 
+  /**
+    * We want to parse log messages. In order to do so, we need a safe way to parse integers in those messages.
+    *
+    * One possibility is to define a function that converts a `String` to an `Option[Int]`
+    *
+    * Hint: You may copy `parseIntSafe` and `tryToOption` from TryExercises
```
Comment (wjlow) [inline src/main/scala/fundamentals/level04/LogParser.scala]: Maybe the hint can be `You've done this before in TryExercises` to force them to think :)

File: src/main/scala/fundamentals/level04/LogParser.scala [hunk 3]
```diff
@@ -62,9 +71,19 @@ object LogParser {
     * scala> parseLog("X blblbaaaaa")
     * = UnknownLog("X blblbaaaaa")
     *
-    * Hint: use the `split` method on `String` and pattern match on `Array`
+    * Here is the beginning of one possible approach.
     **/
-  def parseLog(str: String): LogMessage = ???
+  def parseLog(str: String): LogMessage = {
+    val fields = str.split(",").toList
+    val logOpt: Option[LogMessage] = fields match {
```
Comment (wjlow) [inline src/main/scala/fundamentals/level04/LogParser.scala]: Can we rename to `optLog`? I think that's the convention we use in the codebase.

File: src/main/scala/fundamentals/level04/LogParser.scala [hunk 4]
```diff
@@ -62,9 +71,19 @@ object LogParser {
...
+      case List("I", timestampStr, message) =>
+        parseIntOption(timestampStr).map(timestamp => ???)
+      case _ => ??? // Add more cases
+    }
+
+    // What should we do if logOpt is None?
```
Comment (wjlow) [inline src/main/scala/fundamentals/level04/LogParser.scala]: Ditto (`optLog`)

File: src/main/scala/fundamentals/level03/TryExercises.scala (hunk 1 above)
Comment (shaun-whitely) [reply src/main/scala/fundamentals/level03/TryExercises.scala]: I couldn't find any mention of `.toOption` in the exercises. But if it's in the slides, then I'll remove the exercise. Also the exercise above is `tryToEither`. Is that also redundant?

Comment (wjlow) [reply src/main/scala/fundamentals/level03/TryExercises.scala]: Oops, in `TryExercises`, they would've done `Try => Either`, not to `Option`. We may have also communicated that there's a `.toOption` along with `.toEither` on `Try`.

On second thought I think your addition is fine.

File: src/main/scala/fundamentals/level04/LogParser.scala (hunk 2 above)
Comment (wjlow) [reply src/main/scala/fundamentals/level04/LogParser.scala]: `tryToOption` isn't in `TryExercises` so might be worth rephrasing.

```suggestion
    * Hint: You may copy `parseIntSafe` from TryExercises and use `tryToOption` from above.
```

File: src/main/scala/fundamentals/level04/LogParser.scala (hunk 3 above)
Comment (wjlow) [inline src/main/scala/fundamentals/level04/LogParser.scala]:
```suggestion
    val optLog: Option[LogMessage] = fields match {
```

File: src/main/scala/fundamentals/level04/LogParser.scala (hunk 4 above)
Comment (wjlow) [inline src/main/scala/fundamentals/level04/LogParser.scala]:
```suggestion
    // What should we do if optLog is None?
```

File: src/main/scala/fundamentals/level04/LogParser.scala (hunk 2 above)
Comment (shaun-whitely) [reply src/main/scala/fundamentals/level04/LogParser.scala]: But `tryToOption` is in `TryExercises`. Would you like me to move it to `LogParser`?

Comment (wjlow) [reply src/main/scala/fundamentals/level04/LogParser.scala]: Oh my I'm sorry, I thought `tryToOption` was here lol.

**Pull request merged**